### DISCUSSION TITLE
ENS NFT Avatar

### DISCUSSION THREAD

**Discussion by o-az** [wevm/viem#1451] [Question] [answered]
does `viem` not handle NFT ENS avatars? 

Trying for `nick.eth`:

https://stackblitz.com/edit/wagmi-dev-viem-6zj4ft?file=index.ts

https://app.ens.domains/nick.eth?tab=records

**Comment by jxom** [comment]
Opensea API seems broken. 

https://api.opensea.io/api/v1/metadata/0x495f947276749Ce646f68AC8c248420045cb7b5e/0x8112316025873927737505937898915153732580103913704334048512380490797008551937

**Reply by jxom** [reply, marked as answer]
(the contract uses `https://api.opensea.io/api/v1/metadata/0x495f947276749Ce646f68AC8c248420045cb7b5e/0x{id}` as the URI)

**Reply by o-az** [reply]
@jxom thanks. 
My guess is the reason it works here https://metadata.ens.domains/mainnet/avatar/nick.eth is because the image is cached.
ENS metadata service https://github.com/ensdomains/ens-metadata-service uses https://github.com/ensdomains/ens-avatar and ens-avatar's demo (https://ensdomains.github.io/ens-avatar/) shows a broken image when I search for `nick.eth`.
